routingapi/api.go

```diff
@@ -69,14 +69,22 @@ func (r *RoutingAPI) makeTcpRouteMapping(route config.Route) (models.TcpRouteMap
 
 	r.logger.Info("Creating mapping", lager.Data{})
 
-	return models.NewTcpRouteMapping(
+	return models.NewSniTcpRouteMapping(
 		routerGroupGUID,
 		uint16(*route.ExternalPort),
+		nilIfEmpty(&route.ServerCertDomainSAN),
 		route.Host,
 		uint16(*route.Port),
 		int(route.RegistrationInterval.Seconds())), nil
 }
 
+func nilIfEmpty(str *string) *string {
```

Comment:
I suppose this is necessary because of: https://github.com/cf-gemfire-org/route-registrar/blob/f381d47070b9088b5d48885840a5a24936fb2fd9/config/config.go#L59

I remember reading your comment in slack that this area was not behaving as expected. IIRC, changing this to:
```
ServerCertDomainSAN  *string
```

And the corresponding bind operation here: https://github.com/cf-gemfire-org/route-registrar/blob/f381d47070b9088b5d48885840a5a24936fb2fd9/config/config.go#L292

Should make this function no longer necessary. 

However, there is a guard condition here preventing the value from being unset:
https://github.com/cf-gemfire-org/routing-release/blob/d6ff2417bda22d8250f6a0834a93530b7e30c6da/jobs/route_registrar/templates/registrar_settings.json.erb#L62

And I am unsure what happens if this value would become nil here: https://github.com/cf-gemfire-org/route-registrar/blob/a236249fd9816830500dfcf4aa13e4b266e6d656/messagebus/messagebus.go#L111

Reply:
While `Route.ServerCertDomainSAN` is never `nil` that parameter to `models.NewSniTcpRouteMapping` needs to be `nil` if the value of `Route.ServerCertDomainSAN` is empty so that it won't include the `TcpRouteMapping.SniHostname` in the JSON message to the API server for backwards compatibility with upstream API server. While our version of the routing-api would gladly eat up the empty string the existence of the field at all might trip up the older api-server. We can test this in the future when I resolve the chore I created related to the nil/empty string issue.